### PR TITLE
feat: Add re-simulation feature

### PR DESCRIPTION
**Pull request by OGPoyraz**
## **Description**

This PR cherry-picks https://github.com/MetaMask/metamask-mobile/pull/12073

**Quality gate failures:**

These are expected as we discussed - all those failures will be addressed in the https://github.com/MetaMask/MetaMask-planning/issues/3584 

**Node version update:**

It's a side effect of the bump to `7.0.1` of` @metamask/rpc-errors`.
https://github.com/MetaMask/rpc-errors/blob/main/package.json#L92

## **Related issues**

Fixes: https://github.com/MetaMask/MetaMask-planning/issues/3380

## **Screenshots/Recordings**

This feature already tested and released.
Please see the original picked PR for recording.

## **Pre-merge author checklist**

- [X] I’ve followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Mobile Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [X] I've completed the PR template to the best of my ability
- [X] I’ve included tests if applicable
- [X] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [X] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: .nvmrc [hunk 1]
```diff
@@ -1 +1 @@
-20.14.0
+20.17.0
```
Comment (OGPoyraz) [inline .nvmrc:1]: Please see the comment here 
https://github.com/MetaMask/metamask-mobile/pull/12110#discussion_r1824198015

File: locales/languages/en.json
Comment (matthewwalsh0) [inline locales/languages/en.json:1]: Accidental formatting?

File: app/util/smart-transactions/index.test.ts
Comment (matthewwalsh0) [inline app/util/smart-transactions/index.test.ts:1]: Some more linting here, should we remove these to minimise the diff?

File: app/components/Views/LedgerConnect/Scan.test.tsx [hunk 2]
```diff
@@ -197,7 +197,7 @@ describe('Scan', () => {
     const onDeviceSelected = jest.fn();
 
     const navigateMock = {
-      navigate: jest.fn().mockImplementation(() => {onDeviceSelected(device2)}),
+      navigate: jest.fn().mockImplementation(() => {onDeviceSelected(device2);}),
```
Comment (matthewwalsh0) [inline app/components/Views/LedgerConnect/Scan.test.tsx:200]: Linting to remove?

**Review by matthewwalsh0 — DISMISSED** [review]

File: .nvmrc (hunk 1 above)
Comment (matthewwalsh0) [reply .nvmrc:1]: Should we include that context directly in the PR description for easier audit?

File: locales/languages/en.json
Comment (OGPoyraz) [reply locales/languages/en.json:1]: I think this is the correct formating but I can revert if we don't feel necessary

File: app/util/smart-transactions/index.test.ts
Comment (OGPoyraz) [reply app/util/smart-transactions/index.test.ts:1]: 👍  removing them now

File: locales/languages/en.json
Comment (OGPoyraz) [reply locales/languages/en.json:1]: Reverted

File: .nvmrc (hunk 1 above)
Comment (OGPoyraz) [reply .nvmrc:1]: Noted

**Review by matthewwalsh0 — APPROVED** [review]

**Review by pedronfigueiredo — APPROVED** [review]

**Review by tommasini — APPROVED** [review]
Okok seems that there is an awesome plan in place to fix the sonar quality gate and this is urgent! LGTM!

**Pull request merged**